### ISSUE TITLE
[Housekeeping] [flytekit] Streamline module loading code

### ISSUE THREAD

**Issue by wild-endeavor** [flyteorg/flyte#2540]
## TL;DR
With all the work behind us in `pyflyte package`, `run`, and `register`, along with the legacy `serialize`, the pyflyte code could probably use some cleanup.

## Details
This ticket includes but shouldn't be construed as to be limited to the following:
* Use the same module loading code between the `run` and `register` commands.  Since `run` has to first `register`, and since it also loads when listing commands in a module, it makes sense to reuse the same code.
* There's a lot helper code in the `flytekit/tools` folder, and kind organized poorly.  Attempt to streamline this.
* Figure out proper handling of the `--pkgs` switch that exists at the top level of `pyflyte`, in the `register` command.  Currently it just raises an error.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes
